plugins/rikki/heroeslounge/models/Match.php

```diff
@@ -87,6 +89,30 @@ class Match extends Model
         ]
     ];
 
+    protected static function boot()
+    {
+        parent::boot();
+        // Check if BackendAuth exists, as its missing in some environments like console commands / cronjobs
+        if (!class_exists('BackendAuth')) {
+            return
+        }
+        $user = BackendAuth::getUser();
+            // Check if the user really exists
+        if (!is_object($user)) {
+            return
+        }
+
+        // Add scope that only shows matches that the user has access to
+        static::addGlobalScope('limit_match_access', function ($builder) use ($user) {
+            $builder->whereHas(
+                'playoff.backend_users',
+                function ($backendUserBuilder) use ($user) {
+                    $backendUserBuilder->where('backend_users.id', $user->id);
```

Comment:
Maybe I am reading this wrong, but the way I understand this is that for all matches that have a playoff assigned, and that playoff has backend_users, the playoff is hidden if the user is not allowlisted there. The problem is that existing users that should have access don't see it. Have you tested that users that already have access don't lose it (and, for completeness, that the frontend isn't impacted, but I think this should work out).
If I am correct here, a solution would be to only apply this filter if the user has the individual_playoffs permission.

Reply:
Hm I did have two browsers open, and thought I checked that the first time I pushed. (I obviously did not test the second time sorry, otherwise the semicolons would have been there). But the code clearly does what you think it does and should only apply this if they don’t have general access. The frontend will not be impacted since the backend auth and user will be undefined.